### PR TITLE
Do something about errors from Rspamd

### PR DESCRIPTION
**Pull request by fatalbanana**

### CODE REVIEW

File: plugins/rspamd.js [hunk 1]
```diff
@@ -159,6 +159,13 @@ exports.parse_response = function (rawData, connection) {
         return;
     }
 
+    if (Object.keys(data).length === 1 && data.error) {
+        connection.transaction.results.add(plugin, {
+            err: 'Rspamd returned error: ' + data.error
+        });
+        return;
```
Comment (msimerson) [inline plugins/rspamd.js]: Why have the 'Rspamd returned error:' prefix?

Comment (fatalbanana) [inline plugins/rspamd.js]: Just to distinguish it as some particular type of error. :)

Comment (fatalbanana) [inline plugins/rspamd.js]: I guess it is not strictly necessary in this context. :)

File: plugins/rspamd.js [hunk 2]
```diff
@@ -159,6 +159,13 @@ exports.parse_response = function (rawData, connection) {
         return;
     }
 
+    if (Object.keys(data).length === 1 && data.error) {
+        connection.transaction.results.add(plugin, {
```
Comment (fatalbanana) [inline plugins/rspamd.js]: Would it be preferred to send errors to connection.logerror or connection.transaction.results.add (or both)? Currently the plugin has some mixed behaviour in this respect.

Comment (msimerson) [inline plugins/rspamd.js]: Errors stored to results are automatically emitted as log error, so that is preferable.

**Pull request merged**